### PR TITLE
Fix make warning on GCC

### PR DESCRIPTION
**Pull request by Yangsx-1**
This pr fixes the warning `make[4]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.` on GCC with `-j/--jobs` parameter larger than 1.

### CODE REVIEW

**Comment by Yangsx-1** [comment]
Is there any way to fix the conflict in ninja build? I'm not sure why it happens. @PragmaTwice @tisonkun @git-hulk

**Comment by PragmaTwice** [comment]
Seems ninja does not support something like `$(MAKE)`, but I current have no idea how to workaround it.

**Comment by PragmaTwice** [comment]
Oh, I think one workaround is to check CMAKE_GENERATOR.

File: cmake/lua.cmake [hunk 1]
```diff
@@ -34,10 +34,17 @@ if(NOT lua_POPULATED)
...
+    add_custom_target(make_lua COMMAND make "CC=${LUA_CXX}" "CFLAGS=${LUA_CFLAGS}" liblua.a
+      WORKING_DIRECTORY ${lua_SOURCE_DIR}/src
+      BYPRODUCTS ${lua_SOURCE_DIR}/src/liblua.a
+    )
+  else()
+    add_custom_target(make_lua COMMAND $(MAKE) "CC=${LUA_CXX}" "CFLAGS=${LUA_CFLAGS}" liblua.a
+      WORKING_DIRECTORY ${lua_SOURCE_DIR}/src
+      BYPRODUCTS ${lua_SOURCE_DIR}/src/liblua.a
+    )
+  endif()
```
Comment (PragmaTwice) [inline cmake/lua.cmake]:
```suggestion
  if(CMAKE_GENERATOR STREQUAL "Ninja")
    set(MAKE_COMMAND make)
  else()
    set(MAKE_COMMAND $(MAKE))
  endif()
  add_custom_target(make_lua COMMAND ${MAKE_COMMAND} "CC=${LUA_CXX}" "CFLAGS=${LUA_CFLAGS}" liblua.a
    WORKING_DIRECTORY ${lua_SOURCE_DIR}/src
    BYPRODUCTS ${lua_SOURCE_DIR}/src/liblua.a
  )
```

**Review by torwig — DISMISSED** [review]
LGTM. Thanks, @Yangsx-1 and @PragmaTwice for your help.

**Review by PragmaTwice — DISMISSED** [review]

**Review by tisonkun — COMMENTED** [review]
What about we set `MAKE_COMMAND` at the top level?

nit: revert whitespace changes.

**Review by mapleFU — DISMISSED** [review]

**Comment by Yangsx-1** [comment]
> What about we set `MAKE_COMMAND` at the top level?
> 
> nit: revert whitespace changes.

done

**Review by PragmaTwice — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]

**Comment by tisonkun** [comment]
Thanks for your contribution @Yangsx-1! Merging...

**Pull request merged**